benches/howzit/src/howzit.rs

```diff
@@ -0,0 +1,212 @@
+use std::ops::RangeInclusive;
+use std::path::PathBuf;
+
+use aptos_types::transaction::TransactionPayload;
+use aptos_framework::BuildOptions;
+use aptos_sdk::{
+    move_types::{
+        identifier::Identifier, language_storage::ModuleId
+        
+    }, rest_client::{Client, FaucetClient}, transaction_builder::TransactionBuilder, types::{chain_id::ChainId, transaction::EntryFunction, LocalAccount}
+};
+use anyhow::Context;
+use std::time::{SystemTime, UNIX_EPOCH};
+use crate::build_and_publish_package;
+use tokio::sync::RwLock;
+use std::sync::Arc;
+use url::Url;
+
+#[derive(Debug, Clone)]
+pub enum Probe {
+    Probe1,
+    Probe2,
+    Probe3
+}
+
+impl Probe {
+
+    // associate constant for the probe range
+    const PROBE_RANGE: RangeInclusive<u32> = 0..=3;
+
+    /// Generates a pseudorandom probe such that lower values are exponentially more likely
+    pub fn generate_exponential<R>(rng : &mut R) -> Self
+        where R : rand::Rng
+    {
+
+        // generate a random number between 0 and 2^PROBE_RANGE.end
+        let max = 2u32.pow(*Self::PROBE_RANGE.end());
+        let random = rng.gen_range(0, max);
+
+        // subtract the random number from 2^PROBE_RANGE.end to invert the distribution
+        let inverse = 2u32.pow(*Self::PROBE_RANGE.end()) - random;
+
+        // take the log base 2 of the result to get the probe
+        // we'll use the leading zeros trick here
+        let probe = 32 - (inverse.leading_zeros() as u32);
+        
+        probe.into()
+
+    }
+
+}
+
+impl From<u32> for Probe {
+    fn from(value: u32) -> Self {
+        let value = (
+            value % Self::PROBE_RANGE.end()
+        ) + Self::PROBE_RANGE.start();
+        match value {
+            1 => Probe::Probe1,
+            2 => Probe::Probe2,
+            3 => Probe::Probe3,
+            _ => Probe::Probe1,
+        }
+    }
+}
+
+impl TryInto<Identifier> for Probe {
+    type Error = anyhow::Error;
+
+    fn try_into(self) -> Result<Identifier, Self::Error> {
+        match self {
+            Probe::Probe1 => Ok(Identifier::new("probe_1")?),
+            Probe::Probe2 => Ok(Identifier::new("probe_2")?),
+            Probe::Probe3 => Ok(Identifier::new("probe_3")?),
+        }
+    }
+}
+pub struct Howzit {
+    pub howzit_package_path : PathBuf,
+    pub wallet : Arc<RwLock<LocalAccount>>,
```

Comment:
Does this really have to have interior mutability?

Reply:
`LocalAccount` ought to be borrowed mutably because it has to coordinate its sequencer number globally.